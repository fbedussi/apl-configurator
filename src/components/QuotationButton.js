const QuotationButton = ({ show, text, clickHandler }) => {
    if (!show) {
        return null;
    }

    return <button 
            className={'quotation-btn ctaBtn ctaBtn--inverse'}
            onClick={clickHandler}
        >
            {text}
        </button>;
};

export default QuotationButton;
