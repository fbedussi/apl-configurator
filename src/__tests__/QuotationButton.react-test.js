import React from 'react';
import QuotationButton from '../components/QuotationButton';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';   

const show = true;
const text = 'button text';
const clickHandler = jest.fn();

test('Quotation button component - show', () => {
  const component = renderer.create(
    <QuotationButton
        show={show}
        text={text}
        clickHandler={clickHandler}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Quotation button component - hide', () => {
  const component = renderer.create(
    <QuotationButton
        show={false}
        text={text}
        clickHandler={clickHandler}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('QuotationButton component - call handler when clicked', () => {
  const component = shallow(
    <QuotationButton
        show={show}
        text={text}
        clickHandler={clickHandler}
    />
  );

  component.simulate('click');
  expect(clickHandler).toBeCalled();
});