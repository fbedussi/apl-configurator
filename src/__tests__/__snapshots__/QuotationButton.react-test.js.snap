// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotation button component - hide 1`] = `null`;

exports[`Quotation button component - show 1`] = `
<button
  className="quotation-btn ctaBtn ctaBtn--inverse"
  onClick={[Function]}
>
  button text
</button>
`;
